# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxUserFontSet.h
# Commit: 8ed7c109c239
# Full Hash: 8ed7c109c239b07a02e7f3139f075c8f1d3ff0a0
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2019-09-20 21:52:25
# Description:
#   Bug 1580690 - Ensure src:local() entries in the user font set are refreshed if the platform font list is rebuilt. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D46580
# ==============================================================================

diff -r 932d3794c5d4 -r 8ed7c109c239 gfx/thebes/gfxUserFontSet.h
--- a/gfx/thebes/gfxUserFontSet.h	Thu Sep 19 17:49:39 2019 +0000
+++ b/gfx/thebes/gfxUserFontSet.h	Fri Sep 20 15:24:33 2019 +0000
@@ -298,6 +298,12 @@
   // rebuild if local rules have been used
   void RebuildLocalRules();
 
+  // Discard any font entries created for src:local(), so that they will
+  // be reloaded next time they're needed. This is called when the platform
+  // font list has changed, which means local font entries that were set up
+  // may no longer be valid.
+  void ForgetLocalFaces();
+
   class UserFontCache {
    public:
     // Record a loaded user-font in the cache. This requires that the